libknet/compress.c

```diff
@@ -184,6 +187,38 @@ static int compress_load_lib(knet_handle_t knet_h, int cmp_model, int rate_limit
 	return 0;
 }
 
+static int compress_lib_test(knet_handle_t knet_h)
+{
+	int savederrno = 0;
+	unsigned char src[KNET_DATABUFSIZE];
+	unsigned char dst[KNET_DATABUFSIZE_COMPRESS];
+	ssize_t dst_comp_len = KNET_DATABUFSIZE_COMPRESS, dst_decomp_len = KNET_DATABUFSIZE;
+
+	memset(src, 0, KNET_DATABUFSIZE);
+	memset(dst, 0, KNET_DATABUFSIZE_COMPRESS);
+
+	/*
+	 * NOTE: we cannot use compress and decompress API calls due to locking
+	 * so we need to call directly into the modules
+	 */
+
+	if (compress_modules_cmds[knet_h->compress_model].ops->compress(knet_h, src, KNET_DATABUFSIZE, dst, &dst_comp_len) < 0) {
+		savederrno = errno;
+		log_err(knet_h, KNET_SUB_COMPRESS, "Unable to compress test buffer. Please check your compression settings: %s", strerror(savederrno));
+		errno = savederrno;
+		return -1;
+	}
+
+	if (compress_modules_cmds[knet_h->compress_model].ops->decompress(knet_h, dst, dst_comp_len, src, &dst_decomp_len) < 0) {
+		savederrno = errno;
+		log_err(knet_h, KNET_SUB_COMPRESS, "Unable to decompress test buffer. Please check your compression settings: %s", strerror(savederrno));
+		errno = savederrno;
+		return -1;
+	}
+
+	return 0;
```

Comment:
Shouldn't we check here that the result of the decompression is `KNET_DATABUFSIZE` zero bytes?

Reply:
Done.. funny I didn´t add it because I thought it would be too paranoid :-)

Reply:
Well, it certainly is. But once we're here... Just for fun, have you seen this [memeqzero](https://github.com/rustyrussell/ccan/blob/master/ccan/mem/mem.c#L92)?

Reply:
I have seen similar implementations but at the end we are not screaming for speed or optimizations in that code path, so I decided to go for something more understandable ;)